odoo/addons/base/models/ir_actions_report.py

```diff
@@ -408,8 +411,18 @@ def _prepare_html(self, html):
             if attribute[0].startswith('data-report-'):
                 specific_paperformat_args[attribute[0]] = attribute[1]
 
-        header = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(header_node), base_url=base_url))
-        footer = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(footer_node), base_url=base_url))
+        header_html = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(header_node), base_url=base_url))
+        footer_html = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(footer_node), base_url=base_url))
+
+        def _append_render_done(node_html):
+            node_root = lxml.html.fromstring(node_html)
+            node_body = node_root.find('.//body')
+            node_body.attrib['onload'] = node_body.attrib.get('onload', '') + ";function done() {window.status='report_rendered';};setTimeout(done,200);"
```

Comment:
is 200 really needed? it seems a little arbitrary, couldn't 0 work if the intention is just to do the action after the current JS event loop

also I think it would be slightly better written:

```js
setTimeout(function () { window.status = 'report_rendered'; }, 200);
```

Reply:
> is 200 really needed? it seems a little arbitrary, couldn't 0 work if the intention is just to do the action after the current JS event loop

didn't work correctly without it I just added 200 because that was the default wait time (I guessed it must've been a good estimate)